### PR TITLE
[JA] Translate guardian 1.0.3

### PR DESCRIPTION
**Pull request by starfish719**
I translated `lessons/libraries/guardian` section.
Please review! @elixirschool/japanese

### CODE REVIEW

File: ja/lessons/libraries/guardian.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+JWTトークンを利用して、アプリケーションのあらゆる部分を認証することができます。
+
+* シングルページアプリケーション
+* Controllers (ブラウザセッション経由)
+* Controllers (APIの認証ヘッダ経由)
+* Phoenix Channels
+* サービス間通信
+* プロセス間通信
+* 第三者アクセス (OAuth)
+* Remember me functionality
```
Comment (starfish719) [inline ja/lessons/libraries/guardian.md]: What is `Remember me functionality`  in Japanese？

maybe:
`アカウント再発行`
`パスワード再発行`
`IDを忘れた方はこちら`

File: ja/lessons/libraries/guardian.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+  plug(Guardian.Plug.LoadResource)
+end
+
+pipeline :ensure_authed_access do
+  plug(Guardian.Plug.EnsureAuthenticated, %{"typ" => "access", handler: MyApp.HttpErrorHandler})
+end
+```
+
+これらのpipelineを使用して、さまざまな認証要件を構成することができます。
+最初のpipelineは、セッション内で最初にトークンを見つけ、次にヘッダーにフォールバックします。
```
Comment (starfish719) [inline ja/lessons/libraries/guardian.md]: `The first pipeline tries to find a token first in the session and then falls back to a header.`

`and then falls back to a header.` mean `ヘッダーを確認します` ?

File: ja/lessons/libraries/guardian.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+end
+
+def delete(conn, params) do
+  jwt = Guardian.Plug.current_token(conn)
+  Guardian.revoke!(jwt)
+  respond_somehow(conn)
+end
+```
+
+ブラウザセッションのログインはhoodの下で `encode_and_sign` を呼び出すので、同じように使うことができます。
```
Comment (starfish719) [inline ja/lessons/libraries/guardian.md]: `The browser session login calls `encode_and_sign` under the hood so you can use them the same way.`

What is `hood` ?

File: ja/lessons/libraries/guardian.md (hunk 1 above)
Comment (marocchino) [reply ja/lessons/libraries/guardian.md]: It is "ログインしたままにする"

File: ja/lessons/libraries/guardian.md (hunk 2 above)
Comment (marocchino) [reply ja/lessons/libraries/guardian.md]: maybe) 見つからなかったらヘッダーを確認する

File: ja/lessons/libraries/guardian.md (hunk 3 above)
Comment (marocchino) [reply ja/lessons/libraries/guardian.md]: under the hood is ボンネットの下(つまり内部的に)

File: ja/lessons/libraries/guardian.md [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+やり残していることが一つあります。
+エラーハンドラは `EnsureAuthenticated` Plugに追加しました。
+これは以下に対応する非常に単純なモジュールです。
+
+* `unauthenticated/2`
+* `unauthorized/2`
+
+これらの関数は両方ともPlug.Connの構造体と、Mapのparamsを受け取り、それぞれのエラーを処理します。
+
+これであなたもPhoenix controllerを利用することができます！
```
Comment (riseshia) [inline ja/lessons/libraries/guardian.md]:
```suggestion
Phoenix controllerを使うこともできます！
```

raw text: You can even use a Phoenix controller!

File: ja/lessons/libraries/guardian.md [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+これが最小の設定なので、まずHTTPリクエストを考えましょう。
+
+## HTTPリクエスト
+Guardianは、HTTPリクエストとの連携を容易にするために多数のPlugを提供しています。
+Plugについては [別レッスン](../../specifics/plug/) で学ぶことができます。
+
+GuardianはPhoenixを必要としませんが、以下の例でPhoenixを使用するのが、最も簡単に試すことができます。
+
+HTTPに接続する最も簡単な方法はrouter経由です。
+GuardianのHTTP接続はすべてPlugに基づいているため、Plugを使用できるところならどこでもこれらを使用できます。
```
[the remark this answers is not in the document]
Comment (starfish719) [reply ja/lessons/libraries/guardian.md]: I have never heard the word "HTTP連携" .

maybe)
- `HTTP接続`
- `HTTP通信`

File: ja/lessons/libraries/guardian.md [hunk 6]
```diff
@@ -0,0 +1,288 @@
+---
+version: 1.0.3
+title: Guardian (Basics)
+---
+
+[Guardian](https://github.com/ueberauth/guardian)は、[JWT](https://jwt.io/)（JSON Web Tokens）に基づく、広く使用されている認証ライブラリです。
+
+{% include toc.html %}
+
+## JWTs
```
[the remark this answers is not in the document]
Comment (starfish719) [reply ja/lessons/libraries/guardian.md]: change to `JWT` :)

File: ja/lessons/libraries/guardian.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (riseshia) [reply ja/lessons/libraries/guardian.md]: Yes, I know, and my point is that, differentiate from this term.

The origin text is like `The easiest way to integrate into HTTP is via the router.`,
I think this is about how to integrate JWT with HTTP protocol, not about HTTP connection.
both `HTTP接続` and `HTTP通信` make readers confused with HTTP connection, doesn't it?

**Comment by riseshia** [comment]
and still WIP?

File: ja/lessons/libraries/guardian.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (starfish719) [reply ja/lessons/libraries/guardian.md]:
```suggestion
GuardianのHTTPとの連携は全てPlugに基づいているため、Plugを使用できるところならどこでもこれらを使用できます。
```

maybe

[the remark this answers is not in the document]
Comment (riseshia) [reply ja/lessons/libraries/guardian.md]: Looks better :+1:
could you update L139, too?

**Comment by starfish719** [comment]
@riseshia @marocchino 
I fix it all review comment.
And remove WIP.
Please review.

**Comment by starfish719** [comment]
@riseshia @marocchino 
It was difficult for me to this translate, But I learned a lot of something from this.
Thank you for your reviews 👍

**Pull request merged**